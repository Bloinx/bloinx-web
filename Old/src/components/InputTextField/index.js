/* eslint-disable no-undef */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable no-unused-vars */
import React from "react";
import PropTypes from "prop-types";
import { Typography } from "antd";

import styles from "./styles.module.scss";

const { Text } = Typography;

function InputTextField({ label, name, value, onChange, placeholder, error }) {
  return (
    <div className={styles.TextField}>
      <Text className={styles.TextFieldLabel}>{label}</Text>
      <input
        id={name}
        key={name}
        className={styles.TextFieldInput}
        value={value}
        onChange={onChange}
        placeholder={placeholder}
        name={name}
      />
      {error && <span className={styles.TextFieldError}>{error}</span>}
    </div>
  );
}

InputTextField.defaultProps = {
  placeholder: "",
  label: "",
  value: "",
  error: null,
  onChange: () => {},
};

InputTextField.propTypes = {
  placeholder: PropTypes.string,
  label: PropTypes.string,
  value: PropTypes.string,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  error: PropTypes.string,
};

export default React.memo(InputTextField);
