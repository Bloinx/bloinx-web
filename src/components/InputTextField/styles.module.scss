@import "../../theme.scss";

.TextField {
  display: flex;
  flex-direction: column;
}

.TextFieldLabel {
  font-size: 12px;
  color: rgba($color: #000000, $alpha: 0.54);
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  margin-bottom: 8px;
}

.TextFieldInput {
  border: none;
  border-bottom: 1px solid $gray-light;
  background-color: transparent;
  color: white !important;
  padding-left: 0px;
  padding-right: 0px;
}

.TextFieldError {
  color: $pink;
  font-size: 12px;
  margin-top: 5px;
}
