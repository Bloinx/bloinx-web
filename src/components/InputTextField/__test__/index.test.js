import React from 'react';
import renderer from 'react-test-renderer';

import InputTextField from '../index';

describe('InputTextField', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <InputTextField
        name="some-name"
        value="some-value"
        label="some-label"
        onChange={jest.fn()}
        placeholder="some-placeholder"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
