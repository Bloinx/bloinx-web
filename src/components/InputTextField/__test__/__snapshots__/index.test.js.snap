// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputTextField renders correctly 1`] = `
<div
  className="TextField"
>
  <span
    className="ant-typography TextFieldLabel"
    style={
      Object {
        "WebkitLineClamp": undefined,
      }
    }
  >
    some-label
  </span>
  <input
    className="TextFieldInput"
    id="some-name"
    name="some-name"
    onChange={[MockFunction]}
    placeholder="some-placeholder"
    value="some-value"
  />
</div>
`;
